### DISCUSSION TITLE
Best way to backup app whilst using external database

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#6351] [Help]
I was wondering what the best way is, at present, for backing up Apps which use an external DB,

Currently I have been downloading the JSON App file after any ammendments and also taking a dump of the external DB every night.
However, if the Docker image was to become compromised and everything is to be lost, would this work as a sufficient back up for restoring the app and also pulling the data.
the main potential issue that I see is relationships between tables not being upheld.

Is this the correct way to back up or is there a better way?

**Comment by melohagan** [comment]
One option may be to add an external database as a service within the container. Then you could backup the container before trying out schema changes: https://docs.docker.com/desktop/backup-and-restore/

Otherwise if there is a risk of external database schema changes breaking your app, then exporting your app and backing up your database with a matching timestamp/id is probably wise.

**Reply by PaulMcF1987** [reply]
Thanks @melohagan There are no risks that I know of.
Just always weary with Docker hosted software essentially being a virtual container, that there may be room for error.

If, for example, something did happen to corrupt my app...

If I import the App Json and upload the database dump, in theory, should that get the app back up and running in the same condition as it was in prior to being corrupted?

**Reply by melohagan** [reply]
Yeah it should do, as long as both were working together at the time you backed up.
The same should also be true of a container with a database service that is backed up.
